test/e2e/specs/block-deletion.test.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * Internal dependencies
+ */
+import {
+	clickBlockAppender,
+	getEditedPostContent,
+	newPost,
+	pressWithModifier,
+	META_KEY,
+} from '../support/utils';
+
+const addThreeParagraphsToNewPost = async () => {
+	await newPost();
+
+	// Add demo content
+	await clickBlockAppender();
+	await page.keyboard.type( 'First paragraph' );
+	await page.keyboard.press( 'Enter' );
+	await page.keyboard.type( 'Second paragraph' );
+	await page.keyboard.press( 'Enter' );
+};
+
+const clickOnBlockSettingsMenuItem = async ( buttonLabel ) => {
+	await expect( page ).toClick( '.editor-block-settings-menu__toggle' );
+	const itemButton = ( await page.$x( `//*[contains(@class, "editor-block-settings-menu__popover")]//button[contains(text(), '${ buttonLabel }')]` ) )[ 0 ];
+	await itemButton.click();
+};
+
+describe( 'block deletion -', () => {
+	describe( 'deleting the third block using the Remove Block menu item', () => {
+		beforeAll( addThreeParagraphsToNewPost );
+
+		it( 'results in two remaining blocks', async () => {
+			await clickOnBlockSettingsMenuItem( 'Remove Block' );
+			expect( await getEditedPostContent() ).toMatchSnapshot();
+		} );
+
+		it( 'positions the caret at the end of the second block as evidenced by typing additional text', async () => {
+			await page.keyboard.type( ' - caret was here' );
+			expect( await getEditedPostContent() ).toMatchSnapshot();
+		} );
+	} );
+
+	describe( 'deleting the third block using the Remove Block shortcut', () => {
+		beforeAll( addThreeParagraphsToNewPost );
```

Comment:
Basically we shouldn't want to see `beforeAll( addThreeParagraphsToNewPost );` as it implies we have multiple tests operating on the same sets of blocks, which presumably are being manipulated by the individual test cases.

It should be a `beforeEach` to ensure that each test case has a consistently fresh (un-tampered) set of blocks to work with.